ddl/ddl.go

```diff
@@ -369,7 +370,10 @@ func (d *ddl) start(ctx context.Context, ctxPool *pools.ResourcePool) {
 		d.workers[addIdxWorker] = newWorker(addIdxWorker, d.store, ctxPool)
 		for _, worker := range d.workers {
 			worker.wg.Add(1)
-			go worker.start(d.ddlCtx)
+			w := worker
+			go util2.WithRecovery(ctx, func(ctx context.Context) {
```

Comment:
We may need another `WithRecovery2` that does not need a `context.Context`.
And if we can set the `ddlCtx` in `newWorker`, it can be written like:
```go
go util2.WithRecovery2(worker.start, util2.WithPanicLogf("[ddl-%s] ddl %s", w, d.uuid), util2.WithPanicMetricInc(metrics.LabelDDL))
```